tests/testObjectiveFactors.cpp

```diff
@@ -113,15 +114,15 @@ TEST(Phase, AddGoals) {
       SimpleSwingTrajectory(stance_point, step, num_swing_steps), id);
   EXPECT_LONGS_EQUAL(num_swing_steps, swing_factors.size());
 
-  // Last goal point should have moved just in front of stance_point
-  auto g = boost::dynamic_pointer_cast<PointGoalFactor>(swing_factors.front());
-  EXPECT(assert_equal<Point3>(stance_point + Point3(0.01, 0, 0.035588),
-                              g->goalPoint(), 1e-5));
+  // // Last goal point should have moved just in front of stance_point
```

Comment:
???

Reply:
this is due to rewriting PointGoalFactor as Expressionfactor loses the member function `goalPoint()`. I now make PointGoalFactor inherit from ExpressionFactor, and re-added the `goalPoint()` function to make these tests still working.